torch/csrc/distributed/rpc/testing/faulty_process_group_agent.cpp

```diff
@@ -16,6 +15,7 @@ FaultyProcessGroupAgent::FaultyProcessGroupAgent(
     c10::intrusive_ptr<::c10d::ProcessGroup> pg,
     int numSendRecvThreads,
     std::chrono::milliseconds rpcTimeout,
+    std::unique_ptr<RequestCallback> cb,
```

Comment:
I made the faulty PG agent take a RequestCallback so that it doesn't have to depend on RequestCallbackImpl (which is Python-only). This is the same approach we're already doing for the PG and TP agents.